tools/run_tests/run_interop_tests.py

```diff
@@ -1042,6 +1042,7 @@ def aggregate_http2_results(stdout):
 # Start interop servers.
 server_jobs = {}
 server_addresses = {}
+num_failures = 0
```

Comment:
to be honest, I am not a big fan of having such a large scope for num_failures variable. I'll need to take another look.

Btw, in the past, I think we had the "exit 1" behavior (it was removed because jenkins recognizes failures from the junit report). Perhaps looking at history of this file will reveal a better way to achieve this?